src/MasterSite.php

```diff
@@ -1504,25 +1504,19 @@ public function p4_core_image_block_render(array $attributes, string $content):
             $image_credit = ' ©' . $image_credit;
         }
 
-        $caption_open = '<figcaption>';
-        $caption_close = '</figcaption>';
-        $caption = explode($caption_open, $content, 2)[1] ?? '';
+        $caption = wp_get_attachment_caption($image_id) ?? '';
 
-        if (empty($credit) || strpos($caption, $image_credit) !== false) {
+        if (empty($credit) || (!empty($caption) && strpos($caption, $image_credit) !== false)) {
             return $content;
         }
 
-        if (strpos($content, $caption_open) !== false) {
-            $content = str_replace($caption_close, esc_attr($image_credit) . $caption_close, $content);
-        } else {
-            $content = str_replace(
-                '</figure>',
-                $caption_open . esc_attr($image_credit) . $caption_close . '</figure>',
-                $content
-            );
-        }
-
-        return $content;
+        return str_replace(
```

Comment:
I'm not 100% sure about what to do when there is no caption but there are credits, currently we show the credits in the frontend but not in the editor... Hopefully during UAT Taylor will let us know but for now I've replicated our implementation